imageprocessor/imageprocessor-plugin/src/main/java/pl/edu/uj/kimage/plugin/model/Image.java

```diff
@@ -29,18 +21,7 @@ public int getHeight() {
     }
 
     public Color getColor(final int x, final int y) {
-        isWithinDimensions(x, y);
-        return data[getPixelIndex(x, y)];
-    }
-
-    private int getPixelIndex(final int x, final int y) {
-        return y * width + x;
-    }
-
-    private void isWithinDimensions(final int x, final int y) {
-        if ((x < 0 || x >= width) || (y < 0 || y >= height)) {
-            throw new IllegalArgumentException("Wrong image pixel coordinates. " +
-                    "X should be between " + 0 + " and " + width + ". Y should be between " + 0 + " and " + height);
-        }
+        int index = y * width + x;
```

Comment:
Yeah, you have right, I missed It. I'll fix this today

-----Wiadomość oryginalna-----
Od: "Damian Krawiec" <notifications@github.com>
Wysłano: ‎2016-‎04-‎26 09:58
Do: "kmisztal/KiMage" <KiMage@noreply.github.com>
DW: "annadobrowolska" <anna90dobrowolska@gmail.com>; "Author" <author@noreply.github.com>
Temat: Re: [kmisztal/KiMage] [#41] Builder for Image class (#45)

In imageprocessor/imageprocessor-plugin/src/main/java/pl/edu/uj/kimage/plugin/model/Image.java:
> @@ -29,18 +21,7 @@ public int getHeight() {
>      }
>  
>      public Color getColor(final int x, final int y) {
> -        isWithinDimensions(x, y);
> -        return data[getPixelIndex(x, y)];
> -    }
> -
> -    private int getPixelIndex(final int x, final int y) {
> -        return y * width + x;
> -    }
> -
> -    private void isWithinDimensions(final int x, final int y) {
> -        if ((x < 0 || x >= width) || (y < 0 || y >= height)) {
> -            throw new IllegalArgumentException("Wrong image pixel coordinates. " +
> -                    "X should be between " + 0 + " and " + width + ". Y should be between " + 0 + " and " + height);
> -        }
> +        int index = y * width + x;
Like image.getColor(-1,-1)
—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub